speedy-recovery-client/src/components/core/App.js

```diff
@@ -157,7 +174,19 @@ class App extends Component {
         const appointments = appointmentResource.map(appointment =>
           fhirMapAppointment(appointment, this.state.fhirVersion)
         );
-        this.setState({ appointments });
+        const patients = this.deleteRepeatingElement(
```

Comment:
Surely this method does some unnecessary work? If the user is a practitioner they don't need to go through all of the practitioners of each appointment as they are the practitioner. Also if the user is a patient they don't need to go through all of the patients of each appointment as they are the patient. This should therefore either be split into two methods or an if statement should be added depending on the user